readdir.go

```diff
@@ -0,0 +1,61 @@
+package hdfs
+
+import (
+	"context"
+	"io"
+
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+const defaultListObjectLimit = 100
+
+type listDirInput struct {
+	rp  string
+	dir string
+
+	continuationToken string
+}
+
+func (i *listDirInput) ContinuationToken() string {
+	return i.continuationToken
+}
+
+func (s *Storage) listDirNext(ctx context.Context, page *types.ObjectPage) (err error) {
+	input := page.Status.(*listDirInput)
+
+	dir, err := s.hdfs.Open(input.rp)
```

Comment:
Maybe we need to store `dir` in `page.Status`?

- If we open dir every time in `listDirNext`, we can only list first 100 files.
- We can close `dir` before returning `IterateDone`

Reply:
- If we open dir every time in listDirNext, we can only list first 100 files.

Maybe we could increase the limit each time until we reach `EOF `? but that would repeat the read of the previous file.
So maybe we could probably use a slicing process to weed out the files before 100 each time. Do you think it would work?